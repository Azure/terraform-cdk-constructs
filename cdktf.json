{
  "language": "typescript",
  "app": "npx ts-node ./src/examples/AzureResourceGroup.ts",
  "terraformProviders": ["Azure/azapi@~> 0.6.0"],
  "projectId": "1b5f16bd-67ac-4af6-a33f-7bcafff141eb",
  "context": {
    "excludeStackIdFromLogicalIds": "true",
    "allowSepCharsInLogicalIds": "true"
  }
}