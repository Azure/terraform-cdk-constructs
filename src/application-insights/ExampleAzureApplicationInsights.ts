import { AzureApplicationInsights } from '.';
import { App, TerraformStack} from "cdktf";
import {ResourceGroup} from "@cdktf/provider-azurerm/lib/resource-group";
import {LogAnalyticsWorkspace} from "@cdktf/provider-azurerm/lib/log-analytics-workspace";
import {AzurermProvider} from "@cdktf/provider-azurerm/lib/provider";

import { Construct } from 'constructs';


const app = new App();
    
export class exampleAzureApplicationInsights extends TerraformStack {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    new AzurermProvider(this, "azureFeature", {
        features: {},
      });

    const rg = new ResourceGroup(this, "rg", {
      location: 'eastus',
      name: 'rg-latest',

    });

    const la = new LogAnalyticsWorkspace(this, "log_analytics", {
        location: 'eastus',
        name: 'appinsight-test',
        resourceGroupName: rg.name,
    });

    new AzureApplicationInsights(this, 'testappi', {
      name: 'la-test',
      location: 'eastus',
      resource_group_name: rg.name ,
      application_type: "web",
      workspace_id: la.id,
    });
  }
}

new exampleAzureApplicationInsights(app, "testAzureApplicationInsights");

app.synth();