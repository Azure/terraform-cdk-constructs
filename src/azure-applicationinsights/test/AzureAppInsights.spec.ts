import { Testing, TerraformStack} from 'cdktf';
import { exampleAzureApplicationInsights} from './ExampleAzureApplicationInsights'
import 'cdktf/lib/testing/adapters/jest';


describe('AzureApplicationInsights-Snapshot', () => {
  it('renders a AzureApplicationInsights and check snapshot', () => {

    const synthed = Testing.synthScope((stack) => {
      new exampleAzureApplicationInsights(stack, "testAzureApplicationInsights");
    });
  
    expect(synthed).toMatchSnapshot();
  });
});


describe("AzureApplicationInsights-Terraform", () => {
  it("check if the produced terraform configuration is valid", () => {
    const app = Testing.app();
    const stack = new TerraformStack(app, "test");
    
    Testing.synthScope((stack) => {
      new exampleAzureApplicationInsights(stack, "testAzureApplicationInsights");
    });

    // We need to do a full synth to validate the terraform configuration
    expect(Testing.fullSynth(stack)).toBeValidTerraform();
  });

  it("check if this can be planned", () => {
    const app = Testing.app();
    const stack = new TerraformStack(app, "test");

    Testing.synthScope((stack) => {
      new exampleAzureApplicationInsights(stack, "testAzureApplicationInsights");
    });

    // We need to do a full synth to plan the terraform configuration
    expect(Testing.fullSynth(stack)).toPlanSuccessfully();
  });
});
