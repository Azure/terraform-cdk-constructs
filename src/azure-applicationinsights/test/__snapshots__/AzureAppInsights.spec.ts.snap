// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AzureApplicationInsights-Snapshot renders a AzureApplicationInsights and check snapshot 1`] = `
"{
}"
`;
